lib/FIR/mixin.h

```diff
@@ -61,7 +60,7 @@ template<typename... Ts> struct SumTypeCopyMixin {
 
 // implementation of a (moveable) product type (tuple)
 template<typename... Ts> struct ProductTypeMixin {
-  template<typename A> ProductTypeMixin(A &&x) : t{std::move(x)} {}
+  ProductTypeMixin(Ts... x) : t{std::forward<Ts>(x)...} {}
```

Comment:
You used rvalue reference arguments in the two cases above, but not here.

Reply:
Right. This seemed to be what the reference was suggesting to do. I should be ok that some initializers of the tuple are moved and some copied as context appropriate.

Reply:
By the same argument as the MaybeMixin, it seems `std::forward` is not required here.

Thanks for the details. I'll play around with this some more.

Reply:
The double ctor call was because the class needed a copy ctor (not the default) that didn't make copies of its arguments (i.e., used const T&).  In testing many permutations, `std::forward` is a good choice here as it allows a tuple to hold references and to be constructed with a mix of copies and moves as specified.  For example, `D d{std::move(e), f, &g, h->q()};`